use/mobile-gaming-example.md

```diff
@@ -0,0 +1,540 @@
+---
+layout: default
+title: "Beam Mobile Gaming Example"
+permalink: /use/mobile-gaming-example/
+---
+
+# Apache Beam Mobile Gaming Pipeline Examples
+
+* TOC
+{:toc}
+
+This section provides a walkthrough of a series of example Apache Beam pipelines that demonstrate more complex functionality than the basic WordCount examples. The pipelines in this section process data from a hypothetical game that users play on their mobile phones. The pipelines demonstrate processing at increasing levels of complexity; the first pipeline, for example, shows how to run a batch analysis job to obtain relatively simple score data, while the later pipelines use Beam's windowing and triggers features to provide low-latency data analysis and more complex intelligence about user's play patterns.
+
+> **Note**: These examples assume some familiarity with the Beam programming model. If you haven't already, we recommend familiarizing yourself with the programming model documentation and running a basic example pipeline before continuing. Note also that these examples use the Java 8 lambda syntax, and thus require Java 8. However, you can create pipelines with equivalent functionality using Java 7.
+
+Every time a user plays an instance of our hypothetical mobile game, they generate a data event. Each data event consists of the following information:
+
+- The unique ID of the user playing the game.
+- The team ID for the team to which the user belongs.
+- A score value for that particular instance of play.
+- A timestamp that records when the particular instance of play happened--this is the event time for each game data event.
+
+When the user completes an instance of the game, their phone sends the data event to a game server, where the data is logged and stored in a file. Generally the data is sent to the game server immediately upon completion. However, users can play the game "offline", when their phones are out of contact with the server (such as on an airplane, or outside network coverage area). When the user's phone comes back into contact with the game server, the phone will send all accumulated game data.
+
+This means that some data events might be received by the game server significantly later than users generate them. This time difference can have processing implications for pipelines that make calculations that consider when each score was generated. Such pipelines might track scores generated during each hour of a day, for example, or they calculate the length of time that users are continuously playing the game—both of which depend on each data record's event time.
```

Comment:
As well as offline, we should also mention delays that can happen in the network at various points, so that data may arrive delayed *and* out of order.